code/game/objects/structures/janicart.dm

```diff
@@ -60,6 +61,22 @@
 		updateUsrDialog()
 		to_chat(user, SPAN_NOTICE("You put [I] into [src]."))
 
+	else if(istype(I, /obj/item/clothing/mask/plunger) && !myplunger)
+		if(!user.unEquip(I, src))
+			return
+		myplunger = I
+		update_icon()
+		updateUsrDialog()
+		to_chat(user, SPAN_NOTICE("You put [I] into [src]."))
+
+	else if(istype(I, /obj/item/gun) && !mygun)
+		if(!user.unEquip(I, src))
+			return
+		mygun = I
+		update_icon()
+		updateUsrDialog()
+		to_chat(user, SPAN_NOTICE("You put [I] into [src]."))
+
 	else if(istype(I, /obj/item/device/lightreplacer) && !myreplacer)
```

Comment:
The copypasta is getting out of hand here. 

We could just have an assoc list at the top of the function here of {valid_type: overlay_icon_state}, check the type of I is in that list and do the rest generically (except for mop/signs which are special cases, but that can be done by checking for them first and ELSE-ing into the generic case) or something.

Annoyingly it's not that straightforward, because we have the 'slots', so I won't hard-block this, but I'm not super happy with pasta encroachment.